clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -112,8 +114,56 @@ PointerVariableConstraint::PointerVariableConstraint(
 
 PointerVariableConstraint::PointerVariableConstraint(DeclaratorDecl *D,
                                                      ProgramInfo &I,
-                                                     const ASTContext &C)
-    : PointerVariableConstraint(D->getType(), D, D->getName(), I, C) {}
+                                                     const ASTContext &C) :
+        PointerVariableConstraint(D->getType(), D, D->getName(),
+                                  I, C) { }
+
+
+// Simple recursive visitor for determining if a type contains a typedef
+// entrypoint is find()
+class TypedefLevelFinder : public RecursiveASTVisitor<TypedefLevelFinder> {
+  public:
+
+    static struct InternalTypedefInfo find(const QualType &QT) {
+      TypedefLevelFinder TLF;
+      QualType tosearch;
+      if (auto TDT = dyn_cast<TypedefType>(QT))
+        tosearch = TDT->desugar();
```

Comment:
Does this desugar one level or multiple levels, i.e., if you have a typedef of a typedef?